### PR TITLE
chore(tests): adds view stack and rendering tests

### PR DESCRIPTION
**Pull request by jeabakker**
Tests all core and plugin views for exestence and syntax errors.

Adds base test cases for testing view output

revived version of #11311

### CODE REVIEW

File: engine/tests/classes/Elgg/Plugins/TranslatorUnitTest.php [hunk 1]
```diff
@@ -14,7 +16,7 @@
 	public function up() {
 		$this->translator = new Translator(_elgg_config());
 		$this->translator->loadTranslations('en');
-		$this->translator->registerTranslations($this->getPath(), false, 'en');
+		$this->translator->registerTranslations($this->getPath(). 'languages/', false, 'en');
```
Comment (jdalsem) [inline engine/tests/classes/Elgg/Plugins/TranslatorUnitTest.php]: add space

File: engine/tests/classes/Elgg/Testing.php [hunk 2]
```diff
@@ -90,6 +90,11 @@ public function getAdmin() {
 			if ($admins) {
 				$admin = array_shift($admins);
 			}
+
+			if (!$admin) {
+				$admin = $this->createOne('user', [], ['admin' => 'yes']);
```
Comment (jdalsem) [inline engine/tests/classes/Elgg/Testing.php]: why not createUser?

File: engine/tests/phpunit/integration/Elgg/Views/GroupViewsRenderingTest.php [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+
+	public function getViewNames() {
+		return [
+			'group/elements/summary',
+			'group/default',
+		];
+	}
+
+	public function getDefaultViewVars() {
+		$group = $this->createOne('group');
```
Comment (jdalsem) [inline engine/tests/phpunit/integration/Elgg/Views/GroupViewsRenderingTest.php]: createGroup?

File: engine/tests/phpunit/integration/Elgg/Views/ListingViewsRenderingTest.php [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+	public function getViewNames() {
+		return [
+			'page/components/list',
+			'page/components/gallery',
+			'page/components/table',
+		];
+	}
+
+	public function getDefaultViewVars() {
+		$object = $this->createOne('object');
```
Comment (jdalsem) [inline engine/tests/phpunit/integration/Elgg/Views/ListingViewsRenderingTest.php]: createObject?

File: engine/tests/phpunit/integration/Elgg/Views/ObjectViewsRenderingTest.php [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+
+/**
+ * @group ViewRendering
+ * @group ViewsService
+ */
+class ObjectViewsRenderingTest extends ViewRenderingTestCase {
+
+	public function getViewNames() {
+		return [
+			//'export/entity',
```
Comment (jdalsem) [inline engine/tests/phpunit/integration/Elgg/Views/ObjectViewsRenderingTest.php]: remove

File: engine/tests/phpunit/integration/Elgg/Views/RiverViewsRenderingTest.php [hunk 6]
```diff
@@ -0,0 +1,38 @@
...
+			'river/relationship/friend/create',
+			'river/user/default/profileiconupdate',
+		];
+	}
+
+	public function getDefaultViewVars() {
+		$item = new \ElggRiverItem((object) [
+			'view' => 'river/elements/layout',
+			'action_type' => 'test',
+			'object_guid' => $this->createOne('object')->guid,
```
Comment (jdalsem) [inline engine/tests/phpunit/integration/Elgg/Views/RiverViewsRenderingTest.php]: createObject / User?

File: engine/tests/phpunit/integration/Elgg/Views/UserViewsRenderingTest.php [hunk 7]
```diff
@@ -0,0 +1,67 @@
...
+			'resources/account/forgotten_password',
+			'resources/avatar/edit',
+			'user/default',
+			'user/default/column',
+			'user/elements/summary',
+		];
+	}
+
+	public function getDefaultViewVars() {
+		$user = $this->createOne('user');
```
Comment (jdalsem) [inline engine/tests/phpunit/integration/Elgg/Views/UserViewsRenderingTest.php]: createUser?

File: engine/tests/phpunit/integration/Elgg/Views/WidgetViewsRenderingTest.php [hunk 8]
```diff
@@ -0,0 +1,52 @@
...
+		foreach ($widget_types as $widget_type) {
+			$views[] = "widgets/$widget_type/content";
+			$views[] = "widgets/$widget_type/edit";
+		}
+
+		return $views;
+	}
+
+	public function getDefaultViewVars() {
+		$widget = $this->createOne('object', [
```
Comment (jdalsem) [inline engine/tests/phpunit/integration/Elgg/Views/WidgetViewsRenderingTest.php]: createObject?

File: engine/views.php [hunk 9]
```diff
@@ -11,7 +11,6 @@
 		 */
 		"jquery.js" => "vendor/bower-asset/jquery/dist/jquery.min.js",
 		"jquery.min.map" => "vendor/bower-asset/jquery/dist/jquery.min.map",
-		"jquery-migrate.js" => "vendor/bower-asset/jquery-migrate/jquery-migrate.min.js",
```
Comment (jdalsem) [inline engine/views.php]: why remove this as part of this PR?

File: views/default/forms/usersettings/save.php [hunk 10]
```diff
@@ -6,6 +6,9 @@
  */
 
 $user = elgg_get_page_owner_entity();
+if (!$user) {
```
Comment (jdalsem) [inline views/default/forms/usersettings/save.php]: instanceof

File: views/default/core/settings/account/notifications.php [hunk 11]
```diff
@@ -1,41 +1,46 @@
...
+<?php
+/**
+ * User settings for notifications.
+ *
+ * @package Elgg
+ * @subpackage Core
+ */
+
+$page_owner = elgg_get_page_owner_entity();
+if (!$page_owner) {
```
Comment (jdalsem) [inline views/default/core/settings/account/notifications.php]: instanceof

File: views/default/river/object/comment/create.php [hunk 12]
```diff
@@ -1,39 +1,47 @@
...
-	'message' => elgg_get_excerpt($comment->description),
-	'summary' => $summary,
-]);
+<?php
+/**
+ * Post comment river view
+ */
+
+$item = $vars['item'];
+/* @var ElggRiverItem $item */
```
Comment (jdalsem) [inline views/default/river/object/comment/create.php]: instanceof

**Review by jdalsem — CHANGES_REQUESTED** [review]

File: engine/views.php (hunk 9 above)
Comment (jeabakker) [reply engine/views.php]: found a js lib which is no longer supplies

**Review by jdalsem — APPROVED** [review]

**Pull request merged**